test/fixed_robinhood_hashtable_test.cpp

```diff
@@ -0,0 +1,609 @@
+    auto [value_idx7, did_emplace7] = map.do_try_emplace(128, 256);
+    EXPECT_EQ(value_idx7, 7);
+    EXPECT_TRUE(did_emplace7);
+
+    EXPECT_EQ(map.bucket_at(8).dist(), 3);
+    EXPECT_EQ(map.bucket_at(8).fingerprint(), 6);
+    EXPECT_EQ(map.bucket_at(8).value_index_, 4);
+    EXPECT_EQ(map.value_at(4), Pair(6, 1000));
+
+    EXPECT_EQ(map.bucket_at(9).dist(), 2);
+    EXPECT_EQ(map.bucket_at(9).fingerprint(), 128);
+    EXPECT_EQ(map.bucket_at(9).value_index_, 7);
+    EXPECT_EQ(map.value_at(7), Pair(128, 256));
+
+    EXPECT_EQ(map.bucket_at(0).dist(), 2);
+    EXPECT_EQ(map.bucket_at(0).fingerprint(), 9);
+    EXPECT_EQ(map.bucket_at(0).value_index_, 2);
+    EXPECT_EQ(map.value_at(2), Pair(9, 123));
+
+    // make sure do_try_emplace can find all the places that these elements have ended up
+    auto [value_idx8, did_emplace8] = map.do_try_emplace(13, 2);
+    EXPECT_EQ(value_idx8, 0);
+    EXPECT_FALSE(did_emplace8);
+    EXPECT_EQ(map.value_at(0), Pair(13, 1));
+
+    auto [value_idx9, did_emplace9] = map.do_try_emplace(66, 2);
+    EXPECT_EQ(value_idx9, 6);
+    EXPECT_FALSE(did_emplace9);
+    EXPECT_EQ(map.value_at(6), Pair(66, 66));
+
+    auto [value_idx10, did_emplace10] = map.do_try_emplace(9, 2);
+    EXPECT_EQ(value_idx10, 2);
+    EXPECT_FALSE(did_emplace10);
+    EXPECT_EQ(map.value_at(2), Pair(9, 123));
+
+    // insert_or_assign works just like do_try_emplace except it _does_ overwrite existing elements
+    auto [value_idx11, did_emplace11] = map.do_insert_or_assign(0, -1);
+    EXPECT_EQ(value_idx11, 8);
+    EXPECT_TRUE(did_emplace11);
+
+    EXPECT_EQ(map.bucket_at(0).dist(), 2);
+    EXPECT_EQ(map.bucket_at(0).fingerprint(), 9);
+    EXPECT_EQ(map.bucket_at(0).value_index_, 2);
+    EXPECT_EQ(map.value_at(2), Pair(9, 123));
+
+    EXPECT_EQ(map.bucket_at(1).dist(), 2);
+    EXPECT_EQ(map.bucket_at(1).fingerprint(), 0);
+    EXPECT_EQ(map.bucket_at(1).value_index_, 8);
+    EXPECT_EQ(map.value_at(8), Pair(0, -1));
+
+    auto [value_idx12, did_emplace12] = map.do_insert_or_assign(13, 2);
+    EXPECT_EQ(value_idx12, 0);
+    EXPECT_FALSE(did_emplace12);
+    EXPECT_EQ(map.value_at(0), Pair(13, 2));
+}
+
+TEST(MapOperations, Search)
+{
+    // same operation sequence as the test above, map is in the same state:
+    // 0 (2, 9) -2-> (9, 123)
+    // 1 (2, 0) -8-> (0, -1)
+    // 2
+    // 3 (1, 43) -3-> (43, 999)
+    // 4 (2, 33) -1-> (33, 42)
+    // 5 (3, 23) -5-> (23, 3232)
+    // 6 (4, 13) -0-> (13, 1)
+    // 7 (2, 66) -6-> (66, 66)
+    // 8 (3, 6) -4-> (6, 1000)
+    // 9 (2, 128) -7-> (128, 256)
+
+    IntIntMap10 map{};
+    map.do_insert_or_assign(13, 1);
+    map.do_insert_or_assign(33, 42);
+    map.do_insert_or_assign(9, 123);
+    map.do_insert_or_assign(43, 999);
+    map.do_insert_or_assign(6, 1000);
+    map.do_insert_or_assign(23, 3232);
+    map.do_insert_or_assign(66, 66);
+    map.do_insert_or_assign(128, 256);
+    map.do_insert_or_assign(0, -1);
+
+    print_map_state(map);
```

Comment:
Should avoid printing by default.
We can leave the function in with `[[maybe_unused]]`, or have a flag inside the `print_map_state()` function to turn it off.